spec/section_spec.rb

```diff
@@ -4,37 +4,56 @@
 
 describe SitePrism::Page do
   describe '.section' do
+    class Page < SitePrism::Page; end
+    class Section < SitePrism::Section; end
+
+    class PageWithAnonymousSection < SitePrism::Page
+      section :anonymous_section, '.section' do |s|
+        s.element :title, 'h1'
+      end
+    end
+
+    class PageWithSection < SitePrism::Page
+      section :section, Section, '.section'
+    end
+
+    context 'second argument is a Class' do
+      subject { PageWithSection.new }
+
+      it 'should create a section using the Class' do
+        expect(subject).to respond_to(:section)
+      end
+    end
+
     context 'second argument is not a Class and a block given' do
-      context 'block given' do
-        it 'should create an anonymous section with the block' do
-          class PageWithSection < SitePrism::Page
-            section :anonymous_section, '.section' do |s|
-              s.element :title, 'h1'
-            end
-          end
-
-          page = PageWithSection.new
-          expect(page).to respond_to(:anonymous_section)
-        end
+      subject { PageWithAnonymousSection.new }
+
+      it 'should create an anonymous section using the block' do
```

Comment:
I noticed that the SitePrism::Page tests at the top and the SitePrism::Section specs at the bottom use different naming styles.  Here tests are named "should create an anonym..." whereas below verbs are used without should (ie: "creates an anonym...")   
There's a big debate about which is better, but since you're editing most of this file anyhow you might consider making them consistent.

Reply:
The ones above could maybe read better to especially around `context`.  Right now you'd get output like: "SitePrism::Page .section second argument is a Class should respond to #section" which while parseable is not the clearest.

Reply:
+1 for consistency, and for making sure the spec output reads plainly.

I come down against `should` and in favor of making strong declarations with it blocks. http://www.betterspecs.org/#should